nixos/modules/services/continuous-integration/github-runners.nix

```diff
@@ -0,0 +1,51 @@
+{ config
+, pkgs
+, lib
+, ...
+}@args:
+
+with lib;
+
+let
+  cfg = config.services.github-runners;
+
+in
+
+{
+  options.services.github-runners = mkOption {
+    default = {};
+    type = with types; attrsOf (submodule { options = import ./github-runner/options.nix args; });
+    example = {
+      runner1 = {
+        enable = true;
+        url = "https://github.com/owner/repo";
+        name = "runner1";
+        tokenFile = "/secrets/token1";
+      };
+
+      runner2 = {
+        enable = true;
+        url = "https://github.com/owner/repo";
+        name = "runner2";
+        tokenFile = "/secrets/token2";
+      };
+    };
+    description = lib.mdDoc ''
+      Multiple GitHub Runners.
+    '';
+  };
+
+  config = {
+    systemd.services = flip mapAttrs' cfg (n: v:
+      let
+        svcName = "github-runner-${n}";
+      in
+        nameValuePair svcName
+        (import ./github-runner/service.nix (args // {
+          inherit svcName;
+          cfg = v;
```

Comment:
I think `cfg.name` should default to `n` in case of multiple runners.

Reply:
I agree that would be good, but I'm not sure how to do it... At this point, haven't the defaults on `cfg` already been filled in, so it will currently be equal to `config.networking.hostName`? AFAIK it's not possible to tell whether the user did that deliberately or if a default was used?

Reply:
At least for the `github-runners` options, I think it's fine to remove the default value for `name` and just use the attribute name.

Reply:
Okay, done in the commits below.

Now in the `github-runners` case, the runner's name gets set to the attr name, which is nicely concise, but there's no way to configure it other than that. In other words, if you define `services.github-runners.foo`, then the runner name is always `foo` and the systemd service name is `github-runner-foo`.

I believe this is fine because the set of valid runner names is (roughly?) the set of valid network hostnames, and any such string can be written as a Nix attr. (Albeit with some syntactic weirdness if you have dots or whatever.)

Reply:
Thanks for adding those changes. I had to chance to review the current status of this PR again. I think we're almost good to go 🙂 

> Now in the `github-runners` case, the runner's name gets set to the attr name, which is nicely concise, but there's no way to configure it other than that. In other words, if you define `services.github-runners.foo`, then the runner name is always foo and the systemd service name is `github-runner-foo`.

I don't think we should ignore the `name` option for `github-runners` definitions altogether. I'd much more prefer having the `name` option default to the attribute's name. In the following example, I'd expect a registration of a new runner (and systemd service) as `bar`, not as `foo`.

```
services.github-runners."foo" = {
  enable = true;
  name = "bar";
  url = "https://github.com/veehaitch/testus-maximus";
  tokenFile = "/run/github-token";
};
```

Currently, this example will silently use `foo`, ignoring `name = "bar"` entirely. I think it should be vice versa: if `name` is set, don't bother with the attribute name of the runner definition.

Reply:
Thanks!